### PR TITLE
Issue #12326 Jetty 12.1.x Infinispan flaky tests

### PR DESCRIPTION
**Pull request by olamy**
- **use Infinispan image 15.0.9, use UUID for cache name**
- **checkstyle**
- **use a single instance via singleton holder**
- **reduce memory usage of Infinispan**

### CODE REVIEW

File: tests/jetty-test-session-common/src/main/java/org/eclipse/jetty/session/test/tools/RemoteInfinispanTestSupport.java [hunk 1]
```diff
@@ -112,16 +115,14 @@ public class RemoteInfinispanTestSupport
...
-            _manager.administration().getOrCreateCache("___protobuf_metadata", (String)null).put("session.proto", content);
-
+            manager.administration().getOrCreateCache("___protobuf_metadata", (String)null).put("session.proto", content);
         }
-
     }
 
     public RemoteInfinispanTestSupport(String cacheName)
     {
         Objects.requireNonNull(cacheName, "cacheName cannot be null");
```
Comment (janbartel) [inline tests/jetty-test-session-common/src/main/java/org/eclipse/jetty/session/test/tools/RemoteInfinispanTestSupport.java:138]: Is the `cacheName` now no longer used?

File: tests/jetty-test-session-common/src/main/java/org/eclipse/jetty/session/test/tools/RemoteInfinispanTestSupport.java [hunk 2]
```diff
@@ -50,35 +49,39 @@ public class RemoteInfinispanTestSupport
...
+
+        private Holder()
         {
+            GenericContainer<?> container = new GenericContainer<>(IMAGE_NAME);
+
+            container.withEnv("USER", "theuser")
+                    .withEnv("PASS", "foobar")
+                    .withEnv("MGMT_USER", "admin")
+                    .withEnv("MGMT_PASS", "admin")
+                    //.withEnv("CONFIG_PATH", "/user-config/config.yaml")
```
Comment (janbartel) [inline tests/jetty-test-session-common/src/main/java/org/eclipse/jetty/session/test/tools/RemoteInfinispanTestSupport.java]: If this isn't being used, then delete the line.

File: tests/jetty-test-session-common/src/main/java/org/eclipse/jetty/session/test/tools/RemoteInfinispanTestSupport.java [hunk 3]
```diff
@@ -50,35 +49,39 @@ public class RemoteInfinispanTestSupport
...
+                    .withEnv("JAVA_OPTIONS", "-Xms64m -Xmx256m -Djgroups.dns.query=infinispan-dns-ping.myproject.svc.cluster.local")
+                    .waitingFor(Wait.forListeningPorts(11222))
+                    .withExposedPorts(4712, 4713, 8088, 8089, 8443, 9990, 9993, 11211, 11222, 11223, 11224)
+                    .withLogConsumer(new Slf4jLogConsumer(INFINISPAN_LOG))
+                    //.withClasspathResourceMapping("/config.yaml", "/user-config/config.yaml", BindMode.READ_ONLY)
```
Comment (janbartel) [inline tests/jetty-test-session-common/src/main/java/org/eclipse/jetty/session/test/tools/RemoteInfinispanTestSupport.java]: If not used, delete the line.

File: tests/jetty-test-session-common/src/main/java/org/eclipse/jetty/session/test/tools/RemoteInfinispanTestSupport.java [hunk 4]
```diff
@@ -50,35 +49,39 @@ public class RemoteInfinispanTestSupport
...
 
-    private static final String INFINISPAN_VERSION = System.getProperty("infinispan.docker.image.version", "14.0.25.Final");
+    private static final String INFINISPAN_VERSION = System.getProperty("infinispan.docker.image.version", "15.0.9.Final");
     private static final String IMAGE_NAME = System.getProperty("infinispan.docker.image.name", "infinispan/server") +
             ":" + INFINISPAN_VERSION;
 
-    private static final GenericContainer<?> infinispan = new GenericContainer<>(IMAGE_NAME);
-
-    static
+    private static final class Holder
```
Comment (janbartel) [inline tests/jetty-test-session-common/src/main/java/org/eclipse/jetty/session/test/tools/RemoteInfinispanTestSupport.java]: What's the benefit of doing it this way vs having `static`s?

**Review by janbartel — CHANGES_REQUESTED** [review]
A couple of questions, then a couple of things to just tidy up.

File: tests/jetty-test-session-common/src/main/java/org/eclipse/jetty/session/test/tools/RemoteInfinispanTestSupport.java (hunk 2 above)
Comment (olamy) [reply tests/jetty-test-session-common/src/main/java/org/eclipse/jetty/session/test/tools/RemoteInfinispanTestSupport.java]: Still a draft and working on it
I was just testing it was working with ci but yes I will do it

**Comment by olamy** [comment]
@janbartel still a draft and working on it

File: tests/jetty-test-session-common/src/main/java/org/eclipse/jetty/session/test/tools/RemoteInfinispanTestSupport.java (hunk 4 above)
Comment (olamy) [reply tests/jetty-test-session-common/src/main/java/org/eclipse/jetty/session/test/tools/RemoteInfinispanTestSupport.java]: easier to read :)

**Review by janbartel — APPROVED** [review]

**Pull request merged**